### DISCUSSION TITLE
Throw damage event for every damage type on Interaction entity

### DISCUSSION THREAD

**Discussion by Ielelele29** [PaperMC/Paper#10586] [Ideas]
### Is your feature request related to a problem?

Many types of damage can't be detected for the Interaction entity.

### Describe the solution you'd like.

Currently the EntityDamageEvent event for the Interaction entity is only thrown if a player hits the entity. I ask that the event be thrown also in all other cases of damage (explosions, arrows...).

### Describe alternatives you've considered.

Use another entity together with the Interaction entity to receive these EntityDamageEvents, but still not suitable for many situations

**Comment by pop4959** [comment]
It seems like this is intentional, the [snapshot article](https://www.minecraft.net/en-us/article/minecraft-snapshot-23w07a) introducing interaction entities mentions that they "records attacks ("left clicks") and interactions ("right clicks")."

Though it is interesting that the NBT for left clicking is "attack" which somewhat implies that other attacks would work. [MC-260763](https://bugs.mojang.com/browse/MC-260763) points out these discrepancies, however it looks like Mojang has not taken a stance on it yet.

**Comment by Ielelele29** [comment]
So no chance to find something in paper that allows to use interaction entities in this way? Even just a boolean in the entity?

**Comment by LadyCailin** [comment]
The mojang issue has been closed as "Works As Intended" so I think that they have taken a stance at this point. It would be super great to re-visit this now, and ideally simulate the rest of these events on the entity.
